### PR TITLE
Watch mode fixes

### LINKED ISSUE #67: Double compile initially on watch
So at this point this is nothing more than an annoyance, but it could cause buggy behavior and should eventually be cleaned up. On watch, the project should only compile once, then open up the browser. Right now it runs two times.

### PR DESCRIPTION
**Pull request by jenius**
- Watch compiles only once, and with cleaner code as well
- Fs plugin now works in watch and compile mode

Closes #67

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -117,3 +99,21 @@ export default class Roots extends EventEmitter {
...
+
+function compileCallback (id, err, stats) {
+  if (err) {
+    return this.emit('error', new RootsError({ id: id, message: err }))
+  }
+
+  // Webpack "soft errors" are classified as warnings in roots. An error is
+  // an error. If it doesn't break the build, it's a warning.
+  const jsonStats = stats.toJson()
+  if (jsonStats.errors.length > 0) {
```
Comment (hhsnopek) [inline lib/index.js]: this could be simplified to `if (jsonStats.errors.length) ...` so long as a number is not `0` it'll run

Comment (jenius) [inline lib/index.js]: Honestly I'm wondering why we need to put them to json in the first place, this was originally pulled from an example somewhere I think. But this isn't new code, it was just moved into a function.

Comment (hhsnopek) [inline lib/index.js]: :+1: - I've seen this snippet somewhere else as well... that's gonna bug me now hah

**Comment by kylemac** [comment]
Well that was interesting...

**Pull request merged**